### PR TITLE
feat: revamp actor build endpoint, js builds -> tar

### PR DESCRIPTION
**Pull request by MasterPtato**
Fixes RVT-4117

### CODE REVIEW

**Comment by MasterPtato** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1398?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1398** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1398?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1398?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1397** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1397?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1393** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1393?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/services/build/src/ops/create.rs [hunk 1]
```diff
@@ -1,35 +1,46 @@
...
 	pub game_id: Option<Uuid>,
 	pub env_id: Option<Uuid>,
 	pub display_name: String,
-	pub image_tag: Option<String>,
-	pub image_file: Option<PrepareFile>,
+	pub content: Content,
 	pub multipart: bool,
 	pub kind: BuildKind,
 	pub compression: BuildCompression,
 	pub tags: HashMap<String, String>,
```
Comment (NathanFlurry) [inline packages/services/build/src/ops/create.rs:21]: tags is unused in this op

File: sdks/fern/definition/actor/builds.yml [hunk 2]
```diff
@@ -100,6 +100,7 @@ types:
   PrepareBuildRequest:
     properties:
       name: string
+      tags: unknown
```
Comment (NathanFlurry) [inline sdks/fern/definition/actor/builds.yml:103]: we can't pass tags in prepare bc it'd mark builds as `current` before the build is complete

**Review by NathanFlurry — CHANGES_REQUESTED** [review]

File: sdks/fern/definition/actor/builds.yml (hunk 2 above)
Comment (MasterPtato) [reply sdks/fern/definition/actor/builds.yml:103]: What do you propose?

Comment (NathanFlurry) [reply sdks/fern/definition/actor/builds.yml:103]: leave it as is. make the client call patch tags after uploaded. we can do something cleaner later.

**Pull request merged**